src/sql/src/plan/lowering.rs

```diff
@@ -296,8 +296,8 @@ impl HirRelationExpr {
                     right,
                     on,
                     kind,
-                } if kind.is_lateral() => {
-                    // A LATERAL join is a join in which the right expression has
+                } if right.is_correlated() => {
```

Comment:
The commit message is no longer correct, since now lowering is smarter than it was.

Reply:
Thanks, fixed that an added a release note.